### PR TITLE
Makes the parser stricter

### PR DESCRIPTION
**Pull request by josejulio**
- Support ISO_DATE_TIME codes  If we get an offset, convert to local UTC
 - No longer expect additionalProperties in the event

The applications that were sending additional fields have fixed their messages, we can now remove this data.
As agreed we are accepting timezones - internally we convert them to UTC when parsing.

### CODE REVIEW

File: insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/validator/TestLocalDateTimeValidator.java [hunk 1]
```diff
@@ -8,16 +8,17 @@
...
 
         assertTrue(validator.matches("2020-07-14T13:22:10.133"));
         assertTrue(validator.matches("2020-07-14T13:22:10"));
 
         // Offsets
-        assertFalse(validator.matches("2011-12-03T10:15:30+01:00[Europe/Paris]"));
+        assertTrue(validator.matches("2011-12-03T10:15:30+01:00[Europe/Paris]"));
         assertFalse(validator.matches("2011-12-03T10:15:30[Europe/Paris]"));
-        assertFalse(validator.matches("2011-12-03T10:15:30+01:00"));
+        System.out.println(validator.getErrorMessageDescription());
```
Comment (gwenneg) [inline insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/validator/TestLocalDateTimeValidator.java:20]: Should this be removed?

**Comment by gwenneg** [comment]
⚠️ We must make sure the apps that were sending invalid actions are deployed on prod **before** this is deployed on prod.

File: insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/validator/TestLocalDateTimeValidator.java (hunk 1 above)
Comment (josejulio) [reply insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/validator/TestLocalDateTimeValidator.java]: Yes!

**Review by gwenneg — APPROVED** [review]
LGTM

**Comment by josejulio** [comment]
> warning We must make sure the apps that were sending invalid actions are redeployed on prod with their fix **before** this is deployed on prod.

Already deployed on prod

**Pull request merged**